### ISSUE TITLE
Issues with Windows

### ISSUE THREAD

**Issue by wendeborn8** [lukaswenzl/astrometry#1]
I have installed astrometry, but seem to have trouble running it on Windows 10. I have tried several commands, to no avail.

First, running `astrometry --help` or `astrometry -h` yields the following:

Created TAP+ (v1.2.1) - Connection:
        Host: gea.esac.esa.int
        Use HTTPS: True
        Port: 443
        SSL Port: 443
Created TAP+ (v1.2.1) - Connection:
        Host: geadata.esac.esa.int
        Use HTTPS: True
        Port: 443
        SSL Port: 443
Program version: 1.2
Traceback (most recent call last):
  File "C:\Users\jwend\anaconda3\Scripts\astrometry-script.py", line 33, in <module>
    sys.exit(load_entry_point('astrometry', 'console_scripts', 'astrometry')())
  File "c:\users\jwend\astrometry\astrometry.py", line 414, in main
    args = parseArguments()
  File "c:\users\jwend\astrometry\astrometry.py", line 405, in parseArguments
    args = parser.parse_args()
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 1768, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 1800, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 2006, in _parse_known_args
    start_index = consume_optional(start_index)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 1946, in consume_optional
    take_action(action, args, option_string)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 1874, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 1044, in __call__
    parser.print_help()
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 2494, in print_help
    self._print_message(self.format_help(), file)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 2478, in format_help
    return formatter.format_help()
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 282, in format_help
    help = self._root_section.format_help()
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 213, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 213, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 213, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 213, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 529, in _format_action
    help_text = self._expand_help(action)
  File "C:\Users\jwend\anaconda3\lib\argparse.py", line 621, in _expand_help
    return self._get_help_string(action) % params
TypeError: %o format: an integer is required, not dict
- - - - -
Additionally, running `astrometry sample_file.fits' in the 'sample_images' directory (or on other fits files in other directories) yields the following:

Created TAP+ (v1.2.1) - Connection:
        Host: gea.esac.esa.int
        Use HTTPS: True
        Port: 443
        SSL Port: 443
Created TAP+ (v1.2.1) - Connection:
        Host: geadata.esac.esa.int
        Use HTTPS: True
        Port: 443
        SSL Port: 443
Program version: 1.2
> Astrometry for sample_file.fits
Traceback (most recent call last):
  File "C:\Users\jwend\anaconda3\Scripts\astrometry-script.py", line 33, in <module>
    sys.exit(load_entry_point('astrometry', 'console_scripts', 'astrometry')())
  File "c:\users\jwend\astrometry\astrometry.py", line 469, in main
    result,_  = astrometry_script(fits_image_filename, catalog=args.catalog, rotation_scaling=0, xy_transformation=args.xy_transformation, fine_transformation=args.fine_transformation,
  File "c:\users\jwend\astrometry\astrometry.py", line 766, in astrometry_script
    observation = find_sources(image, vignette,vignette_rectangular,cutouts, sigma_threshold_for_source_detection, FWHM=FWHM)
  File "c:\users\jwend\astrometry\astrometry.py", line 163, in find_sources
    apertures = CircularAperture(positions, r=4.)
  File "C:\Users\jwend\anaconda3\lib\site-packages\photutils\aperture\circle.py", line 148, in __init__
    self.positions = positions
  File "C:\Users\jwend\anaconda3\lib\site-packages\photutils\aperture\attributes.py", line 68, in __set__
    raise ValueError('Input positions must be an (x, y) pixel '
ValueError: Input positions must be an (x, y) pixel position or a list or array of (x, y) pixel positions, e.g., [(x1, y1), (x2, y2), (x3, y3)].
- - - - -
Does astrometry work for Windows 10? If so, what might my issue be? If not, is compatibility with Windows expected to be implemented?

**Comment by wendeborn8** [comment]
I was able to solve the `astrometry -h` issue. I needed to make the '%'  into '%%' on lines 378 and 395 in astrometry.py. Haven't figured out the issues with actually performing the astrometry correction, though.

**Comment by wendeborn8** [comment]
Hmm. It looks like I'm actually having the same issues in Windows 10, MacOS, and Linux. I am installing via:

`git clone https:/github.com/lukaswenzl/astrometry.git`
`pip install -e astrometry/`

**Comment by syfu95** [comment]
I found a way to fix the second issue. 
Edit line 162 and line 769 of astrometry.py:
```
# line 162
positions = (sources['xcentroid'], sources['ycentroid']) 
# line 769
positions = (observation['xcenter'], observation['ycenter'])
```
The shape of `positions` is (2,n), i.e., [(x1,x2,...),(y1,y2,...)], we need to modify its shape to (n,2), i.e., [(x1, y1), (x2, y2), (x3, y3),...]. 
Here are my solutions:
Replace the above two lines by
```
# line 162
positions = []
    for i in range(len(sources)):
        positions.append((sources['xcentroid'][i],sources['ycentroid'][i]))

# line 769
positions = []
    for i in range(len(observation)):
        positions.append((observation.iloc[i,1], observation.iloc[i,2]))
```
Wow, It works! :)

**Comment by lukaswenzl** [comment]
Thank you very much for raising this issue. I was able to reproduce the error and your solutions does work. 
This was caused by a change in photutils and also affects other parts of the code. The linked pull request implements the fix for all of them.

I also implemented the fix for the --help output error by using %% as suggested.

PS: sorry for the later reply it seems my notifications for this repo were not turned on...
